rust/connlib/libs/common/src/session.rs

```diff
@@ -215,6 +215,8 @@ where
         portal_url: impl TryInto<Url>,
         token: String,
         device_id: String,
+        log_dir: String,
```

Comment:
```suggestion
        log_dir: &Path,
```

With this we let the type system do the path sanity check :)

Reply:
If we still need an owned type use [PathBuf](https://doc.rust-lang.org/std/path/struct.PathBuf.html)

Reply:
Gotcha -- yeah it looks like `tracing` wraps this for us so we may not need it for the `file_logger`. We'll probably need for the `log_uploader` module when I implemented that.

Reply:
Updated to use PathBuf